assets/js/modules/analytics-4/components/widgets/EngagedTrafficSourceWidget.test.js

```diff
@@ -16,12 +16,34 @@
  * limitations under the License.
  */
 
-import { render } from '../../../../../../tests/js/test-utils';
+import { freezeFetch, render } from '../../../../../../tests/js/test-utils';
+import {
+	coreKeyMetricsEndpointRegExp,
+	setupRegistryKeyMetricsWidgetHidden,
+	setupRegistryKeyMetricsWidgetNotHidden,
+} from '../../../../util/key-metrics';
 import EngagedTrafficSourceWidget from './EngagedTrafficSourceWidget';
 
 describe( 'EngagedTrafficSourceWidget', () => {
-	it( 'should render the widget', () => {
-		const { getByText } = render( <EngagedTrafficSourceWidget /> );
+	it( 'should not render anything when isKeyMetricsWidgetHidden is not loaded', () => {
```

Comment:
Because all the KM widgets are barren at the moment, this tests looks repetitive and useless, however once they are populated, this may be deemed useful or removed if they don't provide any value.